airflow/providers/github/hooks/github.py

```diff
@@ -45,17 +46,20 @@ def __init__(self, github_conn_id: str = default_conn_name, *args, **kwargs) ->
         self.get_conn()
 
     def get_conn(self) -> GithubClient:
-        """
-        Function that initiates a new GitHub connection
-        with token and hostname ( for GitHub Enterprise )
-        """
+        """Function that initiates a new GitHub connection with token and hostname (for GitHub Enterprise)."""
         if self.client is not None:
             return self.client
 
         conn = self.get_connection(self.github_conn_id)
         access_token = conn.password
         host = conn.host
 
+        # Currently the only method of authenticating to GitHub in Airflow is via a token. This is not the
```

Comment:
Adding this check to align with the [connection documentation](https://airflow.apache.org/docs/apache-airflow-providers-github/stable/connections/github.html#configuring-the-connection) as well.